crates/transaction-pool/src/error.rs

```diff
@@ -213,6 +214,7 @@ impl InvalidPoolTransactionError {
     /// context of the transaction pool and warrants peer penalization.
     ///
     /// See [PoolError::is_bad_transaction].
+    #[allow(clippy::match_same_arms)]
```

Comment:
why do we need an allow here?

Reply:
Because there is documentation that seemed useful to me in each of the branches so I said to myself that for this specific case it was perhaps better to keep them separate. It seems to me that this only concerns 1 or 2 cases maximum in the code.